### PR TITLE
[PF-692] Phase 3 regionality release (support Montreal buckets)

### PR DESCRIPTION
**Pull request by wnojopra**
Follow up from #2680. Phase 3 allows the UI to create workspace buckets in northamerica-northeast1 (Montreal). 

To test:
1) Create a new workspace. Verify that the location dropdown now includes Montreal. Create a workspace in Montreal.
2) Attempt to clone the new workspace. Verify the location dropdown includes US multi-regional, Montreal, and Iowa. Verify the egress warning appears properly when choosing a different location. Create a cloned workspace.
3) Repeat step 2 but for a workspace in the US. 

image

### CODE REVIEW

File: src/components/NewWorkspaceModal.js [hunk 1]
```diff
@@ -118,8 +118,8 @@ const NewWorkspaceModal = Utils.withDisplayName('NewWorkspaceModal', ({
     Ajax(signal).Groups.list().then(setAllGroups),
     !!cloneWorkspace && Ajax(signal).Workspaces.workspace(namespace, cloneWorkspace.workspace.name).checkBucketLocation(cloneWorkspace.workspace.googleProject, cloneWorkspace.workspace.bucketName)
       .then(({ location }) => {
-        // For current phased regionality release, we only allow US workspace buckets.
-        setBucketLocation(isUSLocation(location) ? location : defaultLocation)
+        // For current phased regionality release, we only allow US or NORTHAMERICA-NORTHEAST1 (Montreal) workspace buckets.
+        setBucketLocation(isUSLocation(location) || location === montrealRegion ? location : defaultLocation)
```
Comment (kyuksel) [inline src/components/NewWorkspaceModal.js]: Nit: Perhaps use 

```suggestion
        setBucketLocation(isUSLocation(location) || _.eq(montrealRegion, value) ? location : defaultLocation)
```

to be consistent with the [other comparison below](https://github.com/DataBiosphere/terra-ui/pull/2709/files#diff-1fcbff2d49cbfcbbbd2e6fc3f78f453a36ae84030cbddfec06885366f9c00bc8R94).

**Review by kyuksel — APPROVED** [review]
LGTM 👍🏼 with a nit inline.

Not necessarily as part of this PR but it'd be nice to add some unit tests for some of the sensitive logic that's been added over last few PRs, now that we have unit test examples in place. See [this PR](https://github.com/DataBiosphere/terra-ui/pull/2702) for details.

File: src/_tests/components/region-common.test.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+const montreal = {
+  label: "northamerica-northeast1 (Montreal)",
+  locationType: "region",
+  value: "NORTHAMERICA-NORTHEAST1",
+}
+
+const mockAvailableBucketRegions = [us, usCentral, montreal]
+
+describe('getRegionInfo', () => {
```
Comment (cahrens) [inline src/_tests/components/region-common.test.js:24]: WooHoo, tests! 🥇

File: src/components/NewWorkspaceModal.js [hunk 3]
```diff
@@ -118,8 +118,8 @@ const NewWorkspaceModal = Utils.withDisplayName('NewWorkspaceModal', ({
     Ajax(signal).Groups.list().then(setAllGroups),
     !!cloneWorkspace && Ajax(signal).Workspaces.workspace(namespace, cloneWorkspace.workspace.name).checkBucketLocation(cloneWorkspace.workspace.googleProject, cloneWorkspace.workspace.bucketName)
       .then(({ location }) => {
-        // For current phased regionality release, we only allow US workspace buckets.
-        setBucketLocation(isUSLocation(location) ? location : defaultLocation)
+        // For current phased regionality release, we only allow US or NORTHAMERICA-NORTHEAST1 (Montreal) workspace buckets.
+        setBucketLocation(isUSLocation(location) || _.eq(montrealRegion, location) ? location : defaultLocation)
```
Comment (cahrens) [inline src/components/NewWorkspaceModal.js]: If we are going to continue to slowly roll out locations, could we have a method like `isSupportedBucketLocation` (in region-common)? That method's implementation would change over time, but usages wouldn't care, and we wouldn't need to do things like expose specific locations outside of region-common.

Comment (wnojopra) [reply src/components/NewWorkspaceModal.js]: Thank you! This is **much** better

**Review by cahrens — APPROVED** [review]

**Pull request merged**